prompt2model/dataset_transformer/prompt_based.py

```diff
@@ -133,20 +109,97 @@ async def generate_responses(transform_prompts):
 
         try:
             loop = asyncio.get_event_loop()
-            responses = loop.run_until_complete(generate_responses(transform_prompts))
+            responses = loop.run_until_complete(
+                generate_responses_async(transform_prompts_batch)
+            )
         except API_ERRORS as e:
             handle_api_error(e)
+            # TODO: What to return here?
+        return responses
+
+    def process_responses(
+        self, responses: list, prompt_spec: PromptSpec
+    ) -> tuple[list[str], list[str]]:
+        """Process the responses received from the API.
+
+        Also write the current set of inputs and outputs to a dump text just in case.
 
+        Args:
+            responses: A list of response strings from the API.
+            prompt_spec: The PromptSpec object containing the instruction and examples.
+
+        Returns:
+            A tuple containing two lists: inputs and outputs.
+            - inputs: A list of transformed input strings.
+            - outputs: A list of transformed output strings.
+        """
+        inputs, outputs = [], []
+        show_sample_flag = True
```

Comment:
can remove this

Reply:
I think its good to show some sample outputs

Reply:
I just meant the flag. since it seems the flag can only be true

Reply:
I agree with Saumya. We can add a hidden "debug" setting (which would conventionally be set to False). But I think it's fine to always set this to true for now